app/events/definitions/IEvent.ts

```diff
@@ -0,0 +1,38 @@
+import { IEDataUpdate } from './data/IEDataUpdate';
+import { IEDataGenesis } from './data/IEDataGenesis';
+import { IEDataMessage } from './data/IEDataMessage';
+
+export type EDataDefinition = IEDataUpdate<EDataDefinition> | IEDataGenesis | IEDataMessage;
```

Comment:
Is the "E" of EDataDefinition for "Event"? I'd suggest writing the word instead as it adds to the code's clarity